src/databricks/sql/client.py

```diff
@@ -501,6 +502,11 @@ def execute(
         """
         if parameters is None:
             parameters = []
+        elif self.thrift_backend._use_legacy_parameters:
```

Comment:
I think we need to add unit tests to cover the behavior. Any new code should have unit tests.

What about an app which can connect to different dbrs (thinking of dbt). This new property can serve as a forcing function but the default behaviors should be allow the client code to continue to function. Would you check the server version and decide to use the server support or the old client support of the server does not support it?
If it is going to be addressed in a separate PR, it's fine.

Reply:
We'll circulate a document internally about this decision. I don't think it should be handled / discussed in the Github code review comments as this isn't great visibility across the team.

Agree that this code should be tested and should not merge without tests.

Reply:
```suggestion
        elif self.thrift_backend._use_inline_params:
```

I'd propose naming this as obviously unsafe / inline since `legacy` isn't indicative of what the change actually _does_.